packages/now-cli/src/util/output/create-output.ts

```diff
@@ -111,31 +112,29 @@ function _createOutput({ debug: debugEnabled = false }: OutputOptions = {}) {
     }
   }
 
-  function setSpinner(message: string, delay: number = 300): StopSpinner {
+  function setSpinner(message: string, delay: number = 300): void {
+    spinnerMessage = message;
     if (debugEnabled) {
       debug(`Spinner invoked (${message}) with a ${delay}ms delay`);
-      let isEnded = false;
-      const stop = (() => {
-        if (isEnded) return;
-        isEnded = true;
-        debug(`Spinner ended (${message})`);
-      }) as StopSpinner;
-      stop.text = message;
-      return stop;
+      return;
     }
-
     if (spinner) {
       spinner.text = message;
     } else {
       spinner = wait(message, delay);
     }
-    return spinner;
   }
 
   function stopSpinner() {
+    if (debugEnabled && spinnerMessage) {
```

Comment:
Is there a case where `spinnerMessage` has a value but `spinner` does not?

I would think this function could be simplified to something like:

```js
function stopSpinner() {
    if (spinner) {
      spinner();
      spinner = null;
      debug(`Spinner stopped (${spinnerMessage})`);
      spinnerMessage = '';
    }
}
```

Reply:
It's cause when debug is enabled, then `spinner` is never set.

Reply:
How about this:

```js
function stopSpinner() {
    debug(`Spinner stopped (${spinnerMessage})`);
    spinnerMessage = '';
    if (spinner) {
      spinner();
      spinner = null;
    }
}
```